### PR TITLE
Add debian:bookworm

### PR DESCRIPTION
**Pull request by sitano**
bookworm distro for different Ruby versions + 2.7.8 support

### CODE REVIEW

**Comment by tianon** [comment]
See also https://github.com/docker-library/golang/pull/456#issuecomment-1498122550

File: versions.json [hunk 1]
```diff
@@ -2,17 +2,35 @@
   "2.7": {
     "sha256": "f22f662da504d49ce2080e446e4bea7008cee11d5ec4858fc69000d0e5b1d7fb",
     "variants": [
+      "slim-bookworm",
+      "bookworm",
       "bullseye",
       "slim-bullseye",
       "buster",
       "slim-buster",
```
[the remark this answers is not in the document]
Comment (sitano) [reply versions.json]: @ojab should I remove buster lines though?

**Comment by sitano** [comment]
I have rebased the branch to fix conflicts.
At the moment it does not compile with apline17 and 18 due to openssl version iirc, but not remember for sure.
`update.sh` breaks versions.json for 2.7, I am not sure if I need to fix this script and what it does at all.

**Comment by ojab** [comment]
I believe the single change should be
```diff
diff --git a/versions.sh b/versions.sh
index 0901026..6848b79 100755
--- a/versions.sh
+++ b/versions.sh
@@ -70,8 +70,8 @@ for version in "${versions[@]}"; do
                        sha256: env.shaVal,
                        variants: [
                                (
+                                       "bookworm",
                                        "bullseye",
-                                       "buster",
                                        empty # trailing comma hack
                                | ., "slim-" + .), # https://github.com/docker-library/ruby/pull/142#issuecomment-320012893
                                (
```
and everything else should be autogenerated by `./update.sh` (don't see any special cases for buster, so nothing else to remove + dunno why `need_openssl` flag is needed). 

(NB: I'm not a maintainer)

**Comment by yosifkit** [comment]
Force pushed to drop out Ruby 2.7 and `buster`; also added same fix as https://github.com/docker-library/python/pull/822#discussion_r1227154811.

**Comment by sitano** [comment]
Thank you for accepting this PR!
I would be surprised how many people still sit on 2.7.x.

**Pull request merged**